### DISCUSSION TITLE
Download and Install Python Interpreter from the Internet

### DISCUSSION THREAD

**Discussion by Werni2A** [pypa/pipenv#5632] [Q&A] [answered]
Hi, everyone.

Depending on the project I'm working on I need different versions of Python that I want to install inside the virtual environment created by pipenv. This works fine with `pipenv --python $VERSION` in case I have the requested Python version installed on my host OS. If the requested version is not installed I download the 'embeddable packge' (e.g. https://www.python.org/ftp/python/3.9.7/python-3.9.7-embed-amd64.zip) from the official Python website, extract it and install it via `pipenv --python $INTERPRETER_PATH_OF_DOWNLOADED_PYTHON`.

I haven't found a command line option to instruct pipenv to automatically download the requested Python version and install it into the virtual environment. This would make my workflow a lot easier. Pardon me if I've overlooked this feature in the documentation but if not, what do you think about this feature?

Any chance this will be implemented?

BR
Dominik

**Comment by kalebmckale** [comment, marked as answer]
I suggest using [`pyenv`](https://github.com/pyenv/pyenv) for handling different Python versions.

Here's the location in `pipenv` documentation: https://pipenv.pypa.io/en/latest/advanced/#automatic-python-installation

**Reply by Werni2A** [reply]
Thank you, this is basically what I was looking for. Unfortunately, `pyenv` doesn't support Windows by itself but [`pyenv-win`](https://github.com/pyenv-win/pyenv-win) does. I have to check out the workflow and play with it for some time. It would be great if `pipenv` could call either `pyenv` or `pyenv-win` depending on the host OS (maybe this already works out of the box, haven't checked it yet).

**Reply by kalebmckale** [reply]
Seems like it was something solved back in 2022: https://github.com/pypa/pipenv/pull/4949

**Reply by matteius** [reply]
This already works out of the box -- I use pyenv-win on Windows with pipenv.
